odoo/addons/base/res/res_users.py

```diff
@@ -492,6 +498,18 @@ def authenticate(cls, db, login, password, user_agent_env):
                     _logger.exception("Failed to update web.base.url configuration parameter")
         return uid
 
+    @classmethod
+    def check_active(cls, uid):
+        """Verifies that the given uid is authorized to log-in and return a boolean according to the authorization"""
+        cr = cls.pool.cursor()
```

Comment:
`with cls.pool.cursor() as cr:` with this you don't need to explicitly close the cursor